tests/framework/client/migrationpolicy.go

```diff
@@ -0,0 +1,85 @@
+package client
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	types "k8s.io/apimachinery/pkg/types"
+	watch "k8s.io/apimachinery/pkg/watch"
+	rest "k8s.io/client-go/rest"
+
+	migrationsv1 "kubevirt.io/client-go/generated/kubevirt/clientset/versioned/typed/migrations/v1alpha1"
+
+	v1alpha1 "kubevirt.io/api/migrations/v1alpha1"
+)
+
+var createdMigrationPolicies = make([]migrationPolicyInfo, 0)
+
+type migrationPolicyInfo struct {
+	name string
+	ctx  context.Context
+}
+
+// migrationPolicies implements MigrationPolicyInterface
+type migrationPolicies struct {
+	client          rest.Interface
+	migrationPolicy migrationsv1.MigrationPolicyInterface
```

Comment:
To avoid having to re-implement all of the interface's, you better [embed](https://travix.io/type-embedding-in-go-ba40dd4264df) the migration policy client.

In other words, if you'd define the struct that way:
```go
type migrationPolicies struct {
	migrationsv1.MigrationPolicyInterface
	client          rest.Interface
}

// newMigrationPolicies returns a MigrationPolicies
func newMigrationPolicies(c *TestMigrationPolicyClient) *migrationPolicies {
	return &migrationPolicies{
		MigrationPolicyInterface: c.client.MigrationPolicies(),
		client:                   c.client.RESTClient(),
	}
}
```

Then you would be able to simply:
```go
var c *migrationPolicies = ...
c.Get()
```

And you wouldn't need to recreate all Get, List, Create, etc methods.

This is also easier to implement but would make sure we're up to date with client's methods :)

Reply:
p.s. you can still override whatever methods you need, like Create()
and obviously the above comment applies to every client your created, not just in this file

Reply:
You're totally right. I had some troubles trying to do that, but now it works. Thank you so much